File: petclinic-java-gradle-react-tailwind-ts/src/main/java/dev/pollito/petclinic_java_gradle_react_tailwind_ts/owner/OwnerMapper.java
package dev.pollito.petclinic_java_gradle_react_tailwind_ts.owner;

import dev.pollito.petclinic_java_gradle_react_tailwind_ts.generated.model.OwnerDTO;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.MappingConstants;
import org.mapstruct.MappingTarget;
import org.mapstruct.ReportingPolicy;

@Mapper(componentModel = MappingConstants.ComponentModel.SPRING, unmappedTargetPolicy = ReportingPolicy.IGNORE)
public interface OwnerMapper {

    OwnerDTO updateOwnerDTO(Owner owner, @MappingTarget OwnerDTO ownerDTO);

    @Mapping(target = "id", ignore = true)
    Owner updateOwner(OwnerDTO ownerDTO, @MappingTarget Owner owner);

}

File: petclinic-kt-gradle-thymeleaf-tailwind-webpack-htmx-ts/src/main/kotlin/dev/pollito/petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts/owner/OwnerMapper.kt
package dev.pollito.petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts.owner

import org.mapstruct.Mapper
import org.mapstruct.Mapping
import org.mapstruct.MappingConstants
import org.mapstruct.MappingTarget
import org.mapstruct.ReportingPolicy

@Mapper(
    componentModel = MappingConstants.ComponentModel.SPRING,
    unmappedTargetPolicy = ReportingPolicy.IGNORE,
)
interface OwnerMapper {

    fun updateOwnerDTO(owner: Owner, @MappingTarget ownerDTO: OwnerDTO): OwnerDTO

    @Mapping(
        target = "id",
        ignore = true,
    )
    fun updateOwner(ownerDTO: OwnerDTO, @MappingTarget owner: Owner): Owner
}
